bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/FragmentImpl.java

```diff
@@ -0,0 +1,184 @@
+import com.adobe.cq.forms.core.components.models.form.FormStructureParser;
+import com.adobe.cq.forms.core.components.models.form.Fragment;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { Fragment.class, ComponentExporter.class },
+    resourceType = { FormConstants.RT_FD_FORM_FRAGMENT_V1 })
+@Exporter(
+    name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class FragmentImpl extends PanelImpl implements Fragment {
+
+    public static final String CUSTOM_FRAGMENT_PROPERTY_WRAPPER = "fd:fragment";
+
+    @OSGiService
+    private SlingModelFilter slingModelFilter;
+
+    @OSGiService
+    private ModelFactory modelFactory;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    private String clientLibRef;
+
+    private Resource fragmentContainer;
+
+    @PostConstruct
+    private void initFragmentModel() {
+        ResourceResolver resourceResolver = resource.getResourceResolver();
+        fragmentContainer = null;
+        String fragmentRef = fragmentPath;
+        if (fragmentPath != null && StringUtils.contains(fragmentPath, "/content/dam/formsanddocuments")) {
+            fragmentRef = GuideUtils.convertFMAssetPathToFormPagePath(fragmentPath);
+        }
+        fragmentContainer = resourceResolver.getResource(fragmentRef + "/" + JcrConstants.JCR_CONTENT + "/guideContainer");
+        if (fragmentContainer != null) {
+            FormStructureParser formStructureParser = resource.adaptTo(FormStructureParser.class);
+            if (formStructureParser != null) {
+                this.clientLibRef = formStructureParser.getClientLibRefFromFormContainer();
+            }
+        }
+    }
+
+    @JsonIgnore
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @JsonIgnore
+    public String getClientLibRef() {
+        return clientLibRef;
+    }
+
+    @Override
+    public @NotNull Map<String, ? extends ComponentExporter> getExportedItems() {
+        if (itemModels == null) {
+            itemModels = getChildrenModels(request, ComponentExporter.class);
+        }
+        return itemModels;
+    }
+
+    protected <T> Map<String, T> getChildrenModels(@Nullable SlingHttpServletRequest request, @NotNull Class<T> modelClass) {
+        Map<String, T> models = new LinkedHashMap<>();
+        List<Resource> filteredChildrenResources = getFilteredChildrenResources(fragmentContainer);
+        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(request) {
+
+            @Override
+            public Object getAttribute(String attrName) {
+                if (FormConstants.RESOURCE_CALLER_PATH.equals(attrName)) {
+                    String resourceCallerPath = (String) super.getAttribute(FormConstants.RESOURCE_CALLER_PATH);
```

Comment:
Please rename the constant to REQ_ATTR_RESOURCE_CALLER_PATH

Reply:
ok